### PR TITLE
[Communication] - Chat - Address requested changes from Azure Review

### PR DESCRIPTION
**Pull request by glorialimicrosoft**
# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [ ] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/master/eng/mgmt/mgmtmetadata).
- [ ] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running generate.ps1. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version,

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

### CODE REVIEW

File: sdk/communication/Azure.Communication.Chat/README.md [hunk 1]
```diff
@@ -193,18 +193,18 @@ Use `GetChatThread` to retrieve a chat thread from the service.
 `threadId` is the unique id of the thread.
 
 ```C# Snippet:Azure_Communication_Chat_Tests_Samples_GetThread
-ChatThreadProperties chatThread = await chatClient.GetChatThreadPropertiesAsync(threadId);
+ChatThreadProperties chatThread = await chatThreadClient.GetPropertiesAsync();
 ```
 
 ### Get threads (for a participant)
 
 Use `GetChatThreadsInfo` to get the list of chat threads for the participant that instantiated the chatClient.
```
Comment (juancamilor) [inline sdk/communication/Azure.Communication.Chat/README.md:201]: Use `GetChatThreads`

File: sdk/communication/Azure.Communication.Chat/api/Azure.Communication.Chat.netstandard2.0.cs [hunk 2]
```diff
@@ -142,19 +151,15 @@ public partial class ChatThreadProperties
...
-        public string Code { get { throw null; } }
-        public System.Collections.Generic.IReadOnlyList<Azure.Communication.Chat.CommunicationError> Details { get { throw null; } }
-        public Azure.Communication.Chat.CommunicationError InnerError { get { throw null; } }
-        public string Message { get { throw null; } }
-        public string Target { get { throw null; } }
-    }
     public partial class CreateChatThreadResult
     {
         internal CreateChatThreadResult() { }
         public Azure.Communication.Chat.ChatThreadProperties ChatThreadProperties { get { throw null; } }
```
Comment (juancamilor) [inline sdk/communication/Azure.Communication.Chat/api/Azure.Communication.Chat.netstandard2.0.cs:157]: Name of the field : ChatThread
public Azure.Communication.Chat.ChatThreadProperties ChatThread { get { throw null; } }

File: sdk/communication/Azure.Communication.Chat/src/ChatThreadClient.cs [hunk 3]
```diff
@@ -371,10 +410,10 @@ public virtual Response<AddChatParticipantsResult> AddParticipant(ChatParticipan
             {
                 return _chatThreadRestClient.AddChatParticipants(Id, new[] { participant.ToChatParticipantInternal() }, cancellationToken);
             }
-            catch (Exception ex)
+            catch (RequestFailedException requestFailedException)
```
Comment (juancamilor) [inline sdk/communication/Azure.Communication.Chat/src/ChatThreadClient.cs:413]: We need to check if there is any invalid participant, and throw the exception if there is :

        public virtual Response<AddChatParticipantsResult> AddParticipant(ChatParticipant participant, CancellationToken cancellationToken = default)
        {
            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(ChatThreadClient)}.{nameof(AddParticipant)}");
            scope.Start();
            try
            {
                Response<AddChatParticipantsResult> addChatParticipantsResult = _chatThreadRestClient.AddChatParticipants(Id, new[] { participant.ToChatParticipantInternal() }, cancellationToken);
                if (addChatParticipantsResult.Value.InvalidParticipants.Count > 0)
                {
                    throw _clientDiagnostics.CreateRequestFailedException(addChatParticipantsResult.GetRawResponse());
                }
            }
            catch (Exception ex)
            {
                scope.Failed(ex);
                throw;
            }
        }

File: sdk/communication/Azure.Communication.Chat/src/ChatThreadClient.cs [hunk 4]
```diff
@@ -352,10 +391,10 @@ public virtual async Task<Response<AddChatParticipantsResult>> AddParticipantAsy
             {
                 return await _chatThreadRestClient.AddChatParticipantsAsync(Id, new[] { participant.ToChatParticipantInternal() }, cancellationToken).ConfigureAwait(false);
             }
-            catch (Exception ex)
+            catch (RequestFailedException requestFailedException)
```
Comment (juancamilor) [inline sdk/communication/Azure.Communication.Chat/src/ChatThreadClient.cs:394]: See comment below

File: sdk/communication/Azure.Communication.Chat/tests/ChatClients/ChatClientsTests.cs [hunk 5]
```diff
@@ -397,7 +397,7 @@ public async Task GetThreadsShouldSucceed()
             ChatClient chatClient = CreateMockChatClient(200, GetThreadsApiResponsePayload);
 
             //act
-            AsyncPageable<ChatThreadItem> chatThreadsInfo = chatClient.GetChatThreadsItemAsync();
+            AsyncPageable<ChatThreadItem> chatThreadsInfo = chatClient.GetChatThreadsAsync();
```
Comment (juancamilor) [inline sdk/communication/Azure.Communication.Chat/tests/ChatClients/ChatClientsTests.cs:400]: chatThreads

**Review by juancamilor — CHANGES_REQUESTED** [review]

File: sdk/communication/Azure.Communication.Chat/README.md [hunk 6]
```diff
@@ -94,7 +94,7 @@ chatThreadClient.UpdateTopic(topic: "Launch meeting");
 
 ### Send a message
 ```C# Snippet:Azure_Communication_Chat_Tests_Samples_SendMessage_KeyConcepts
-string messageId = chatThreadClient.SendMessage("Let's meet at 11am");
+string messageId = chatThreadClient.SendMessage("Let's meet at 11am").Value.Id;
```
[the remark this answers is not in the document]
Comment (glorialimicrosoft) [reply sdk/communication/Azure.Communication.Chat/README.md]: changed

File: sdk/communication/Azure.Communication.Chat/README.md (hunk 1 above)
Comment (glorialimicrosoft) [reply sdk/communication/Azure.Communication.Chat/README.md]: changed

File: sdk/communication/Azure.Communication.Chat/api/Azure.Communication.Chat.netstandard2.0.cs (hunk 2 above)
Comment (glorialimicrosoft) [reply sdk/communication/Azure.Communication.Chat/api/Azure.Communication.Chat.netstandard2.0.cs]: changed

File: sdk/communication/Azure.Communication.Chat/src/ChatThreadClient.cs (hunk 3 above)
Comment (glorialimicrosoft) [reply sdk/communication/Azure.Communication.Chat/src/ChatThreadClient.cs]: Fixed, question: would AddParticipant still return a success result if there was no error? or it should not return anything?

File: sdk/communication/Azure.Communication.Chat/tests/ChatClients/ChatClientsTests.cs (hunk 5 above)
Comment (glorialimicrosoft) [reply sdk/communication/Azure.Communication.Chat/tests/ChatClients/ChatClientsTests.cs]: changed

File: sdk/communication/Azure.Communication.Chat/src/ChatThreadClient.cs (hunk 4 above)
Comment (juancamilor) [reply sdk/communication/Azure.Communication.Chat/src/ChatThreadClient.cs]: Same 
public virtual Task<Response> AddParticipant(ChatParticipant participant, CancellationToken cancellationToken = default)

**Review by juancamilor — APPROVED** [review]

File: sdk/communication/Azure.Communication.Chat/src/ChatThreadClient.cs (hunk 3 above)
Comment (juancamilor) [reply sdk/communication/Azure.Communication.Chat/src/ChatThreadClient.cs]: my bad, It should not return anything:
public virtual Response AddParticipant(ChatParticipant participant, CancellationToken cancellationToken = default)

**Review by juancamilor — APPROVED** [review]

**Pull request merged**